### PR TITLE
OMID-194 OmidTableManager cannot create commit and timestamp tables i…

### PR DESCRIPTION
**Pull request by chrajeshbabu**
…n kerberos cluster(Rajeshbabu)

### CODE REVIEW

**Comment by stoty** [comment]
Let me recap what I see:

The original problem was that the old code didnt't check for a pre-existing kerberos ugi, and forced logging in with the provided credentials, even if they weren't provided.

This made Phoenix almost unusable in a secure environment, as every principal would have needed their own specialized client config, with principals and keytabs.

The new code fixes it by reusing existing credentials from the cache / UGI, and for good measure replaces most of the kerberos login code which the Phoenix version.

**Review by stoty — APPROVED** [review]
+1 LGTM

Can you also add some tests in an addendum / follow up ticket ? 
SecureUserConnectionsIT in Phoenix could probably be adopted easily.

**Review by joshelser — APPROVED** [review]
> SecureUserConnectionsIT in Phoenix could probably be adopted easily.

+1 for that

Looks like the code for EmbeddedPhoenixDriver was easily adapted here. That's been working well in Phoenix, so it should be good to go here.

For clarity, will this code work if I just do...
```
$ kinit jelser@MYREALM
Password: *********
$ java my.fancy.phoenix.omid.App
```
.. but don't have the hbase client principal/keytab configuration set? If it doesnt' work, that's fine, but we should be clear about what is expected to work (e.g. on http://phoenix.apache.org/omid.html)

**Comment by chrajeshbabu** [comment]
> .. but don't have the hbase client principal/keytab configuration set? If it doesnt' work, that's fine, but we should be clear about what is expected to work (e.g. on http://phoenix.apache.org/omid.html)

It's working Josh.

**Comment by chrajeshbabu** [comment]
> SecureUserConnectionsIT in Phoenix could probably be adopted easily.

Will raise separate issue for test cases and add those.

File: hbase-common/src/main/java/org/apache/omid/tools/hbase/HBaseLogin.java [hunk 1]
```diff
@@ -29,27 +33,95 @@
...
+                                hbaseConfig.get(SecureHBaseConfig.HBASE_CLIENT_KEYTAB_KEY));
+                        UserGroupInformation.setConfiguration(hbaseConfig);
+                        User.login(hbaseConfig, SecureHBaseConfig.HBASE_CLIENT_KEYTAB_KEY, SecureHBaseConfig.HBASE_CLIENT_PRINCIPAL_KEY, null);
+                        LOG.info("Successful login to secure cluster");
                     }
                 }
             } else {
-                LOG.info("User {}, already trusted (Kerberos). Avoiding 2nd login as it causes problems", ugi.toString());
+                // The user already has Kerberos creds, so there isn't anything to change in the ConnectionInfo.
+                LOG.debug("Already logged in as {}", currentUser);
```
[the remark this answers is not in the document]
Comment (chrajeshbabu) [reply hbase-common/src/main/java/org/apache/omid/tools/hbase/HBaseLogin.java:67]: > Nit: logging the UGI even when the config doesn't contain kerberos principal/keytab may be beneficial.
This is already there

else {
            LOG.warn("Security NOT enabled when connecting to HBase. Act at your own risk. NULL UGI returned");
        }

[the remark this answers is not in the document]
Comment (stoty) [reply hbase-common/src/main/java/org/apache/omid/tools/hbase/HBaseLogin.java:67]: I meant the case when we use kerberos with the pre-existing cache.
This branch is gated by **&& credsProvided**, so we do not log this when we're already loggied into kerberos, and no credentials were provided (the default case)

**Comment by chrajeshbabu** [comment]
merged the changes.

**Pull request closed without merging**